pkg/endpointmanager/host.go

```diff
@@ -29,17 +31,28 @@ func (mgr *endpointManager) HostEndpointExists() bool {
 }
 
 // OnAddNode implements the endpointManager's logic for reacting to new nodes
-// from K8s. It is currently not implemented as the endpointManager has not
-// need for it. This adheres to the subscriber.NodeHandler interface.
-func (mgr *endpointManager) OnAddNode(node *slim_corev1.Node,
+// from K8s.
+// This adheres to the subscriber.NodeHandler interface.
+func (mgr *endpointManager) OnAddNode(newNode *slim_corev1.Node,
 	swg *lock.StoppableWaitGroup) error {
 
+	node.SetLabels(newNode.GetLabels())
+
+	nodeEP := mgr.GetHostEndpoint()
+	if nodeEP == nil {
+		// if host endpoint does not exist yet, labels will be set when it'll be created.
+		return nil
+	}
+
+	newLabels := labels.Map2Labels(newNode.GetLabels(), labels.LabelSourceK8s)
+	nodeEP.UpdateLabels(context.TODO(), newLabels, newLabels, false)
```

Comment:
I'm not 100% sure if we shouldn't filter these labels here?
Looking at `OnUpdateNode` runs labels through `labelsfilter.Filter` before triggering `ModifyIdentityLabels`.

Reply:
Fixed. Also, looking at the `OnUpdateNode` handler, I realized that the info labels should not be changed, so I set the related `nodeEP.UpdateLabels` parameter to `nil`.